[FE] 프로젝트 1주차 리뷰요청

sungik-choi opened the pull request:
## 작업내용

- 개발환경 구축
- 라우팅 페이지 설계
- 이슈 리스트 컴포넌트 설계

sungik-choi committed:
[FE/#2] Docs : README 파일 삭제

sungik-choi committed:
[FE/#2] Docs : .gitignore 파일 추가

sungik-choi committed:
[FE/#2] Feat : 필수 패키지 설치

- babel
- webpack
- react

sungik-choi committed:
[FE/#2] Feat : webpack 기본 환경설정 파일 생성

Elllin committed:
[FE/#2] Feat : resolve 추가, .env 설치

Elllin committed:
[FE/#2] Feat : dotenv -> dotenv-webpack 으로 변경 설치

Elllin committed:
[FE/#2] Feat : cleanwebpack plugin prod에 적용

sungik-choi committed:
[FE/#2] Feat : terser-webpack-plugin, webpack-merge 설치

sungik-choi committed:
[FE/#2] Feat : terser-webpack-plugin 설정

sungik-choi committed:
[FE/#2] Feat : webpack-merge로 설정 파일 분리

Elllin committed:
[FE/#2] Feat : split chunks 추가 및 소스맵 추가

Elllin committed:
[FE/#2] Feat : package에서 script 명령어 설정

Elllin committed:
[FE/#2] Feat : alias 기본 구조 추가, resolve에 modules 삭제

sungik-choi committed:
[FE/#2] Fix : git rebase 문제 수정

sungik-choi committed:
[FE/#2] Chore : 테스트용 index 파일 생성

sungik-choi committed:
[FE/#2] Chore : .gitignore dist 폴더 추가

sungik-choi committed:
[FE/#2] Fix : webpack prod 에러 수정

- splitChunks 를 TereserPlugin 바깥으로 이동
- CleanWebpackPlugin 사용을 npm 문서에 작성돼있는 방식대로 변경
- CleanWebpackPlugin 생성자 내부에 dist 삭제. 있으면 작동하지 않음

sungik-choi committed:
[FE/#2] Feat : webpack common babel-loader 설정

- output은 파일명만 지정해주므로 삭제

sungik-choi committed:
[FE/#2] Chore : loader 옵션 위치 변경

- options 보다 위로 오게

sungik-choi committed:
[FE/#2] Feat : eslint 패키지 설치

sungik-choi committed:
[FE/#2] Feat : eslint, prettier 설정 파일 추가

sungik-choi committed:
[FE/#2] Style : eslint에 맞게 코드 컨벤션 변경

Elllin committed:
[FE/#2] Feat : 폴리필 설치 및 common.js에 적용

Elllin committed:
[FE/#2] Feat : proptypes 설치

Elllin committed:
[FE/#2] Feat : app.js 추가 기본 구조 구성

Elllin committed:
[FE/#2] Feat : index.js 파일 생성 및 기본 구조 생성

Elllin committed:
[FE/#2] Feat : index.html에 id값 root 생성

Elllin committed:
[FE/#2] Chore : index.jsx 파일 index.js파일로 수정

sungik-choi committed:
Merge pull request #16 from codesquad-member-2020/fe/feature/init

[FE] 개발환경 구성

sungik-choi committed:
[FE/#5] Feat : Route 페이지 구조 설정

sungik-choi committed:
[FE/#5] Feat : Issues 페이지 컴포넌트 구조 작성

sungik-choi committed:
[FE/#5] Feat : 공용 Header 컴포넌트 생성

Elllin committed:
Merge pull request #19 from codesquad-member-2020/fe/feature/design

Elllin committed:
[FE/#20] Feat : Material UI 설치

sungik-choi committed:
Merge pull request #25 from codesquad-member-2020/fe/feature/materialUiInstall

[FE/#20] Material UI 설치

Elllin committed:
[FE/#3] Fix : webpack설정 text 수정

jsx 문법 안되는 오류 수정

Elllin committed:
[FE/#3] feat : 글로벌 스타일 추가

reset, button reset 추가

Elllin committed:
[FE/#3] Feat : styled-reset 추가

글로벌 스타일 적용 및 reset설치

Elllin committed:
[FE/#3] Feat : material ui icons 설치

Elllin committed:
[FE/#3] Feat : 북마크 아이콘 추가, header 스타일

header 배경색, 기본 layout

Elllin committed:
[FE/#3] Feat : 북마크 아이콘 위치 조정

타이틀과 높이 맞춤

Elllin committed:
Merge branch 'fe/dev' into fe/feature/headerUi

sungik-choi committed:
Merge pull request #27 from codesquad-member-2020/fe/feature/headerUi

[FE] Header UI

sungik-choi committed:
[FE/#8] Feat : font-family 추가

- Mac 에서도 시스템 폰트를 잘 표현할 수 있도록 폰트 패밀리 변경

sungik-choi committed:
[FE/#8] Feat : webpack 절대경로 추가

- Componets, Pages, Styles 절대경로 추가

sungik-choi committed:
[FE/#8] Feat : eslint-import-resolver-alias 설치 및 절대경로 변경

- eslint 에서도 절대경로를 잘 잡아주기 위한 플러그인 설치
- eslint 에도 절대경로 설정
- 절대경로 이름 변경 (이름 앞에 @)

sungik-choi committed:
[FE/#8] Feat : Mock 파일 추가

- 절대경로 추가

sungik-choi committed:
[FE/#8] Chore : ESLint rules 추가

- https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md
- react-eslint : prop-types에서 별도로 기본 props값을 설정해주라는 경고가 발생하는데, 구조분해할당을 통해 기본값을 줄 것이므로 삭제함

sungik-choi committed:
[FE/#8] Feat : GlobalStyles button 삭제

- Material UI 와 충돌 문제로 삭제

sungik-choi committed:
[FE/#8] Feat : 재사용 CustomTable 컴포넌트 생성

- material-ui 사용
- headContents와 bodyContents를 props로 전달받아 사용하는 table 틀 제작

sungik-choi committed:
[FE/#8] Feat : CustomTable 수정

- Flex Box 추가
- propTypes : isRequired 추가

sungik-choi committed:
[FE/#8] Feat : IssueTable 제작 중

- 로직 없음
- CustomTable을 사용해 표 렌더링하도록 제작중

sungik-choi committed:
[FE/#8] Feat : @material-ui/lab 설치

- Avatar group 을 사용하기 위해 설치

sungik-choi committed:
[FE/#8] Feat : CustomTable flex 속성 변경

- Avatar Group과 내용간의 정렬을 위해 space-between 설정 추가

sungik-choi committed:
[FE/#8] Feat : Avatar Group, Chip, Icon, Link 추가

- Link : 이슈명
- Chip : 이슈 레이블
- Avatar, Avatar Group : Asignee 리스트

sungik-choi committed:
[FE/#8] Style : CustomTable 파일 위치 변경

- 폴더 삭제 후 .jsx 파일만 컴포넌트 디렉토리 내부로 이동

Elllin committed:
[FE/#10] Feat : popup-state설치 및 popup 버튼 구현

material ui popup 버튼 삽입

sungik-choi committed:
[FE/#8] Feat : eslint rules 추가

- 스타일 코드를 컴포넌트 코드 아래에 선언해도 에러가 발생하지 않도록 규칙 추가

sungik-choi committed:
[FE/#8] Feat : IssueTable 컴포넌트 분리

- IssueList 내용들 각 컴포넌트로 분리

sungik-choi committed:
[FE/#8] Fix : 불필요한 CSS 속성 제거

Elllin committed:
[FE/#10] Feat : 필터 버튼 스타일 수정, 아이콘 추가

스타일 디폴트로 수정, 다운아이콘 추가

sungik-choi committed:
[FE/#8] Feat : CustomTable 테이블 바디 key 지정

- key를 bodyContents의 id 값으로 지정
- PropTypes를 더 상세하게 명시
- IssueTable : 바뀐 방식으로 데이터 생성 후 CustomTable에 전달해서 렌더링

Elllin committed:
[FE/#10] Refactor : 스타일 인라인 추가에서 useStyles이용

sungik-choi committed:
[FE/#8] Feat : Label 재사용 가능 컴포넌트로 변경

- 컴포넌트 상위 디렉토리로 이동
- name, backgroundColor, color를 받아 별도의 컴포넌트로 사용할 수 있도록 변경
- 기본 매개변수는 defaultProps로 사용

sungik-choi committed:
[FE/#8] Feat : default props 설정 재변경

- https://stackoverflow.com/questions/41488537/react-defaultprops-vs-es6-default-params-when-destructuring-performances-issu
- defaultProps를 쓰는 편이 리액트에서 기본 매개변수 타입 체킹을 진행해주기 때문에 더 좋다고 한다.
- 기존에 설정해놨던 lint 설정을 다시 on 시키고, custom table의 기본 매개변수도 defaultProps로 재설정했다.

sungik-choi committed:
[FE/#8] Feat : Title 컴포넌트 props 받게 변경

sungik-choi committed:
[FE/#8] Feat : Details 컴포넌트 props 받게 변경

sungik-choi committed:
[FE/#8] Fix : id 타입 변경

- string -> number

sungik-choi committed:
[FE/#8] Style : 재사용 컴포넌트 common 디렉토리 내부로 이동

sungik-choi committed:
[FE/#8] Feat : 유저 목 데이터 생성

sungik-choi committed:
[FE/#8] Feat : AvatarStack props로 렌더링하도록 변경

Elllin committed:
[FE/#10] Feat : header 제외 width 80%로 조절

Elllin committed:
[FE/#10] Refactor : header padding 제거

Elllin committed:
[FE/#10] Feat : 팝업창 크기 조절 및 hover시 background color 효과

Elllin committed:
[FE/#10] Feat : 글로벌 스타일 폰트사이즈 변경

Elllin committed:
[FE/#10] Feat : nav 위치 변경  margin 추가

Elllin committed:
[FE/#10] Feat : 필터 버튼 border 추가 및 변수 지정

버튼 border, popup background color 변수로 분리

Elllin committed:
[FE/#10] Refactor : popup 전체 사이즈 및 폰트 사이즈 변경

Elllin committed:
[FE/#10] Refactor : menu list를 다른 컴포넌트로 분리

Elllin committed:
[FE/#10] Feat : menuList title추가

Elllin committed:
[FE/#10] Refactor : menu list 중복 렌더 컴포넌트 없앰

title일 경우 따로 렌더링 하는 함수를 구현했지만 중복이라 렌더 함수
하나로 변수에 변하는 값을 넣어서 처리

Elllin committed:
[FE/#10] Chore : menu list 재사용 가능함으로 common 폴더로 이동

Elllin committed:
[FE/#10] Refactor : menuList title에 들어갈 text를 props를 통해 받음

text값은 상수로 분리

Elllin committed:
[FE/#10] Refactor : menuList사용법 주석 추가

sungik-choi committed:
[FE/#8] Refactor : border-color 로 변경

- 2px solid 는 불필요한 override

sungik-choi committed:
[FE/#8] Refactor : isRequired 속성 추가

Elllin committed:
[FE/#10] Refactor : app styled component 대신 container사용

sungik-choi committed:
[FE/#8] Feat : 커스텀 테이블 테두리 추가

sungik-choi committed:
[FE/#8] Feat : 체크박스 갯수 로직 추가

- table 헤더 selected 텍스트 추가 (전체 체크박스 수)
- useState를 사용해 체크박스 로직 추가

Elllin committed:
[FE/#10] Refactor : menuList에 key값 추가

sungik-choi committed:
[FE/#8] Style : ESLint 삼항연산자 규칙 추가

sungik-choi committed:
[FE/#8] Refactor : IssueList -> Issue로 파일명 변경

- 개별 이슈를 IssueList로 이름짓는 건 어울리지 않는 거 같아 Issue로 네이밍 변경
- 구조상 issueList를 별개로 두는 것은 어려워보인다. CustomTable 내부에서 map 함수를 통해 뿌려주고 있기 때문에...

sungik-choi committed:
[FE/#8] Refactor : selected -> selectedIssue 로 변경

- 의미를 더 담기 위해 state명 변경

sungik-choi committed:
[FE/#8] Feat : issue prop-types 설정

sungik-choi committed:
[FE/#8] Refactor : 함수명 변경

- 영어 어순에 맞게

sungik-choi committed:
[FE/#8] Refactor : 테이블 헤더 내용 Toolbar 파일 분리

sungik-choi committed:
[FE/#8] Feat : Toolbar prop-types 설정

Elllin committed:
[FE/#10] Refactor : navigation에 nav태그 box로 변경

sungik-choi committed:
[FE/#8] Feat : SelectedNumber, TotalCheckbox 파일 분리

- TotalCheckBox -> 소문자 box로 변경

sungik-choi committed:
[FE/#8] Refactor : SelectednNumber, TotalCheckbox 삭제

- 파일로 분리하기에 너무 작다고 생각됨. 괜히 Props만 더 불필요하게 전달하는 느낌이 들어 삭제함.

sungik-choi committed:
[FE/#8] Feat : 이슈 상태 아이콘 변경 로직 추가

Elllin committed:
[FE/#10] Refactor : styled components 사용을material ui로 변경

sungik-choi committed:
Merge pull request #35 from codesquad-member-2020/fe/feature/filterMenu

[FE] Filter Menu Ui 수정

Elllin committed:
[FE/#4] input 창 생성

Elllin committed:
[FE/#4] Feat : search bar filter버튼 flex, search bar 크기 조정

Elllin committed:
[FE/#4] Refactor : search bar, filter버튼 border radius조절

Elllin committed:
[FE/#4] Refactor : placeholder 변경 및 search bar 색상 변경

Elllin committed:
[FE/#4] Refactor : searchBar 분리 컴포넌트를 합침

primarySearchAppBar컴포넌트를 searchBar컴포넌트 내에서 return

sungik-choi committed:
Merge pull request #36 from codesquad-member-2020/fe/feature/searchBar

[FE] Search bar UI

sungik-choi committed:
Merge branch 'fe/dev' of https://github.com/codesquad-member-2020/issue-tracker-05 into fe/feature/table

# Conflicts:
#	FE/package.json
#	FE/src/App.js

Elllin committed:
Merge pull request #38 from codesquad-member-2020/fe/feature/table

Elllin committed:
[FE/#6] Feat : 링크버튼 생성 및 import

Elllin committed:
[FE/#6] Feat : 버튼 컴포넌트 활용하여 레이블,마이스톤 버튼 생성

Elllin committed:
[FE/#6] Feat : button group으로 두 버튼 하나로 묶음`

Elllin committed:
[FE/#6] Refactor : searchBar 높이 조정

Elllin committed:
[FE/#6] Refactor : LinkButtons 필요없는 코드 제거

Elllin committed:
[FE/#6] Refactor : 버튼 대문자 -> 소문자로 변경

Elllin committed:
[FE/#6] Feat : LinkButton 컴포넌트 propTypes추가

Elllin committed:
[FE/#40] Refactor : menu컴포넌트 스타일 컴포넌트 부분과 합침

Elllin committed:
[FE/#40] Refactor : menuList protoTypes추가

Elllin committed:
[FE/#40] Refactor : menu 컴포넌트 alias로 import 개선

sungik-choi committed:
Merge pull request #41 from codesquad-member-2020/fe/feature/refactor

[FE] 필터, 메뉴 컴포넌트 리팩토링

sungik-choi committed:
Merge pull request #39 from codesquad-member-2020/fe/feature/linkButtons

[FE] 레이블, 마일스톤 버튼 UI

Elllin committed:
[FE/#7] Feat : new Issue 버튼 생성 및 import

Elllin committed:
[FE/#7] Feat : new issue버튼 스타일

Elllin committed:
[FE/#7] Refactor : nav 전체 width 조금씩 변경

searchbar width 늘림, 버튼들 양옆 padding 추가

sungik-choi committed:
Merge pull request #42 from codesquad-member-2020/fe/feature/newIssueButton

[FE] 이슈 생성 버튼 UI

sungik-choi committed:
[FE/#37] Style : Prettier 설정에 맞게 코드 스타일 변경

sungik-choi committed:
[FE/#37] Feat : PropTypes isRequired 추가

sungik-choi committed:
[FE/#37] Refactor : Header 텍스트 분리

sungik-choi committed:
[FE/#37] Feat : Filter Button 구현

- Toolbar의 Button, Menu 부분 분리 후 Menu 는 삭제
- 누르면 나오는 메뉴를 제외한 버튼 컴포넌트 자체만 구현
- 반복을 피하기 위해 FilterButton 컴포넌트 새로 생성

sungik-choi committed:
[FE/#37] Feat : MenuList 기본 Props 추가

- 둘중 하나만 있으므로 isRequired 삭제
- 기본 매개변수 빈 문자열로 추가

sungik-choi committed:
[FE/#37] Refactor : 변수명 변경

sungik-choi committed:
[FE/#37] Refactor : MenuList key값 변경

- 인덱스를 사용하지 않고 text를 그대로 키값으로 변경
- 사용하지 않는 theme 삭제

sungik-choi committed:
[FE/#37] Refactor : grey 색상 theme에서 가져오도록 변경

- makeStyles의 theme.palette에 접근해서 가져오도록 변경

sungik-choi committed:
[FE/#12] Chore : ESLint 속성 추가

- https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md
- popupState 라이브러리와 충돌나서 무시

Elllin committed:
Merge pull request #43 from codesquad-member-2020/fe/feature/filters

sungik-choi committed:
[FE/#12] Refactor : Filter 스타일 수정

- left 로 줄에 맞게 변경
- 기본값인 default 제거 (주입하지 않아도 기본값)

sungik-choi committed:
[FE/#12] Feat : FilterButton 드롭다운 메뉴 버튼으로 변경

- TableFilters 에서 만든 객체 배열을 FilterButton에 넣고, FilterButton 내부에서 Popover와 MenuList를 통해 드롭다운 메뉴 렌더링

sungik-choi committed:
[FE/#12] Feat : MenuList clickHandler 추가

sungik-choi committed:
[FE/#12] Feat : CustomAvatar 추가

- 필터 메뉴의 사람 목록에서도 아바타를 쓰기위해 별도의 크기 조정된 커스텀 아바타 컴포넌트 추가

sungik-choi committed:
[FE/#12] Feat : AvatarStack 커스텀 아바타로 변경

sungik-choi committed:
[FE/#12] Feat : FilterButton 클릭 핸들러 추가

sungik-choi committed:
[FE/#12] Feat : MenuList 옵션 추가

- 필터에 사용하기 위해 clickHandler, avatar, name 옵션 추가
- Typograhy API에 있는 속성을 활용해 스타일 변경
- 주석 제거

sungik-choi committed:
[FE/#12] Feat : MenuList 색상 수정

- 역할에 맞는 색으로 수정
- 주석 제거

sungik-choi committed:
[FE/#12] Feat : FilterButton MenuList props 변경

- 아바타, 이름 정보는 옵션값으로 넘기게 변경

sungik-choi committed:
[FE/#12] Feat : FilterButton Mock Data

- 아바타, 이름을 보기 위한 목 데이터 추가

sungik-choi committed:
[FE/#12] Feat : FilterButton PropTypes 설정

sungik-choi committed:
[FE/#12] Feat : PropTypes 수정

sungik-choi committed:
[FE/#12] Feat : MenuList label 추가

- 재사용 컴포넌트 Label을 사용해서 옵션으로 렌더링하도록 추가

sungik-choi committed:
[FE/#12] Feat : label prop-types 추가

sungik-choi committed:
[FE/#12] Feat : filterlist mock 데이터로 이동

sungik-choi committed:
[FE/#12] Refactor : 불필요한 코드, 파일 삭제

- 코드 스타일 일관성 있게

sungik-choi committed:
[FE/#12] Feat : Header Material UI Appbar 사용하도록 변경

- 코드 스타일 일관성 있게 Material UI 로

sungik-choi committed:
[FE/#12] Refactor : Proptypes 리팩토링

Elllin committed:
Merge pull request #44 from codesquad-member-2020/fe/feature/filterMenu

Elllin committed:
[FE/#11] Feat : clear button 생성

search bar tag form으로 변경, aria-label 삽입

Elllin committed:
[FE/#11] Feat : button 스타일

색상, hover effect

Elllin committed:
[FE/#11] Refactor : 버튼 text 색상 진하게 변경

Elllin committed:
[FE/#11] Feat : clearbutton 조건부 렌더링 추가

menulist text추가

crongro commented on FE/package.json:
```diff
@@ -0,0 +1,51 @@
+  "main": "index.js",
+  "scripts": {
+    "start": "webpack-dev-server --open --config webpack.dev.js",
+    "build": "webpack --config webpack.prod.js"
```
일주일마다 데모를 위해서가 아니더라도, 
그 주의 릴리즈를 한다고 생각하고, production build 를 해서 배포본에 포함해서 데모도 하고 해보시죠.

crongro commented on FE/src/components/Issues/IssueTable/Issue/AvatarStack.jsx:
```diff
@@ -0,0 +1,41 @@
+  },
+}));
+
+AvatarStack.propTypes = {
```
좋은시도에요. typescript 를 쓰지 않고도 리액트컴포넌트는 props에 대해서는 이렇게 파라미터에 대해서 타입을 정의할 수 있죠.

crongro commented on FE/src/components/Issues/IssueTable/IssueTable.jsx:
```diff
@@ -0,0 +1,60 @@
+import Issue from "./Issue/Issue";
+
+const IssueTable = () => {
+  const [selectedIssue, setSelectedIssue] = useState(new Set());
```
Set을 썼군요.
Array 차이를 알아두고 설명할 수 있어야 합니다.
이런  비슷한 자료구조를 비교할때는 성능적인 차이점은 무엇인지 알아두면 더 좋고요.

crongro commented on FE/src/components/Issues/IssueTable/IssueTable.jsx:
```diff
@@ -0,0 +1,60 @@
+  const isSelectedIssue = (id) => selectedIssue.has(id);
+  const isAtLeastOneSelectedIssue = (issueList) =>
+    selectedIssueSize > 0 && selectedIssueSize < issueList.length;
+  const isAllSelectedIssue = (issueList) =>
```
복잡한  연산자를 이렇게 함수로 만들어 두니 보기 좋은거 같네요.

crongro commented on FE/src/components/Issues/Navigation/LinkButtons/LinkButton.jsx:
```diff
@@ -0,0 +1,44 @@
+  );
+};
+
+const useStyles = makeStyles(() => ({
```
useStyles 을 써서 좀더 깔끔해진거 같네요. 
👍

crongro commented on FE/src/components/common/Label.jsx:
```diff
@@ -0,0 +1,40 @@
+  }),
+});
+
+Label.defaultProps = {
```
다음엔 typescript 를 써보는 게 좋겠네요.
이런 장점을 다 포함하기도하고, 지금도 잘 구현중이니 ts도  소화가능하실 듯.

crongro reviewed (approved):
components/common 에 있는 컴포넌트를 서비스내에서 재사용해보면서 범용성을 올리면 좋겠네요. 

material 을 빠르게 잘 이해하고 적용했네요. useStyle등을 잘 활용한 점도 좋네요.

컴포넌트 단위는 지금 수준이 매우 좋습니다.

상태관리 시작하면 useReducer  , Context API잘 활용해보세요.

crongro commented:
@sungik-choi 리뷰요청을 제가 놓친거 같네요.
구현이 잘  되어 있어서 별로 리뷰드릴 것도 없네요. 👍

Elllin committed:
[FE/#11] Refactor : app에서 불러온 컴포넌트 page로 이동

navigation, clearButton 컴포넌트 page- issues로 이동

sungik-choi committed:
[FE/#45] Feat : Avatar 툴팁 추가

- 아바타에 마우스를 hover하면 해당 유저의 id 표시

sungik-choi committed:
[FE/#45] Feat : LinkButton MaterialUI의 기본값으로 변경

- count는 Chip 컴포넌트를 사용
- Button의 startIcon으로 아이콘 사용

Elllin committed:
[FE/#11] Refactor : useStyles 제거

sungik-choi committed:
[FE/#45] Feat : NewIssueButton, LinkButton 분리

- NewIssueButton은 삭제하고, 재사용을 위해 PrimaryButton을 따로 생성 후, 컴포넌트를 사용하도록 변경
- LinkButton은 재사용할 여지가 있으므로 재사용 컴포넌트로 위치이동

Elllin committed:
[FE/#11] Chore : 잘못 올라간 파일 삭제

sungik-choi committed:
[FE/#45] Feat : Menu, SearchBar UI 수정

- Material UI에서 가능한 부분들 교체

sungik-choi committed:
[FE/#45] Feat : CustomTable 마지막 요소 borderbottom 제거

sungik-choi committed:
[FE/#45] Feat : Label에 Pointer-event: none 추가

sungik-choi committed:
[FE/#45] Fix : AvatarStack 동작 수정

- Buttongroup과 같이 props를 넘겨줘야 제대로 작동함.
- width, height를 AvatarStack에서도 지정해줘야 제대로 동작함.

sungik-choi committed:
[FE/#45] Feat: Filter Menu 간격 추가

sungik-choi committed:
[FE/#45] Style : 불필요한 부분 제거, 개행

sungik-choi committed:
[FE/#45] Style : 초기 보여주기 데이터 변경

sungik-choi committed:
[FE/#45] Feat : LinkButton count pointer-event: none 추가

sungik-choi committed:
[FE/#45] Style : 불필요한 스타일 코드 제거 및 버튼 radius 수정

sungik-choi committed:
[FE/#45] Fix : PropTypes 오류 수정

Elllin committed:
Merge pull request #47 from codesquad-member-2020/fe/feature/ui-unify

Elllin committed:
Merge branch 'fe/dev' into fe/feature/clearButton

sungik-choi committed:
Merge pull request #46 from codesquad-member-2020/fe/feature/clearButton

[FE] 초기화 버튼 UI

sungik-choi committed:
[FE/#63] Fix : 병합 충돌 수정

- mockArr -> menuList

sungik-choi committed:
[FE/#63] Fix : Box 2개인 부분 수정

- Box의 component를 form으로해서 하나로 병합

sungik-choi committed:
[FE/#63] Fix : div -> nav

- mb: 2로 ClearButton과 통일

sungik-choi committed:
[FE/#65] Feat : moment 설치

sungik-choi committed:
[FE/#65] Feat : 시간 차이 계산 유틸 함수 작성

sungik-choi committed:
[FE/#65] Feat : Utils 절대경로 추가

sungik-choi committed:
[FE/#65] Feat : 시간 계산 함수 적용

sungik-choi committed:
[FE/#66] Feat : tooltip props에 따라 툴팁 출력 여부 설정

sungik-choi committed:
[FE/#66] Feat : Custom Avatar 툴팁 속성 추가

sungik-choi committed:
[FE/#66] Fix : 오타 수정

sungik-choi committed:
[FE/#66] Fix : 잘못된 import 제거

sungik-choi committed:
[FE/#66] Feat : 헤더에 아바타 버튼 추가

Elllin committed:
Merge pull request #64 from codesquad-member-2020/fe/fix

sungik-choi committed:
[FE/#66] Feat : 로그아웃 메뉴 추가

sungik-choi committed:
[FE/#66] Feat : UserMenu 컴포넌트 분리

sungik-choi committed:
[FE/#66] Feat : 조건부 렌더링 추가

- 스타일 약간 수정

sungik-choi committed:
[FE/#66] Feat : PropTypes 추가

sungik-choi committed:
[FE/#66] Chore : 주석 추가

Elllin committed:
Merge pull request #67 from codesquad-member-2020/fe/feature/headerAvatar

Elllin committed:
Merge pull request #68 from codesquad-member-2020/fe/feature/time

sungik-choi committed:
[FE/#69] Fix : Util 함수 기본 반환값 추가

sungik-choi committed:
[FE/#69] Chore : reducers, contexts 절대경로 추가

sungik-choi committed:
[FE/#69] Feat : issuesReducer, issuesContext 생성

sungik-choi committed:
[FE/#69] Feat : Issues 기본 상태관리 적용

sungik-choi committed:
[FE/#69] Feat : userContext, userReducer 추가

Elllin committed:
[FE/#70] Feat : 파일 구조설계 - 1차

1차 구조설계(수정가능)

Elllin committed:
[FE/#70] Feat : 기본 컴포넌트  구조 작성

Elllin committed:
[FE/#70] Feat : 타이틀 컴포넌트 import, title data추가

mock데이터로 렌더링 , 스타일 작업은 미완성

Elllin committed:
[FE/#70] Feat : 타이틀 layout, margin 추가

Elllin committed:
[FE/#70] Feat : edit button 생성

Elllin committed:
[FE/#70] Feat : 타이틀, edit 버튼 layout

양쪽 끝으로 layout

Elllin committed:
[FE/#70] Feat : input box 생성

Elllin committed:
[FE/#70] Feat : edit버튼 토글기능

버튼 클릭시 input창과 타이틀 번갈아가며 렌더링

Elllin committed:
[FE/#70] Feat : title 내용 새로운 컴포넌트로 분리

Elllin committed:
[FE/#70] Feat : 타이틀 수정시 타이틀박스에 반영

Elllin committed:
[FE/#70] Feat : save, close버튼 기능 구현

save, close버튼을 inputBox로 이동, edit버튼을 TitleBox로 이동

Elllin committed:
[FE/#70] Refactor : save,close버튼 layout수정

input과flex되도록 변경

Elllin committed:
[FE/#70] Feat : props로 전달 함수들 useCallback 사용

Elllin committed:
[FE/#70] Refactor : save버튼 submit으로 변경

form에서 submit으로 처리, 필요없는button컴포넌트 삭제

Elllin committed:
[FE/#70] input field 공통 컴포넌트로 분리 / 재사용

Elllin committed:
[FE/#70] Feat : inputBox aria label추가

Elllin committed:
[FE/#70] Feat : status icon, button추가

status icon common으로 이동하여 재사용

Elllin committed:
[FE/#70] Feat : inputBox propsType 추가,input 값 없을때 저장 안됨

Elllin committed:
[FE/#70] Feat : refactor : props type 수정

titleBox, inputBox, handler isRequired 추가

Elllin committed:
[FE/#70] Feat : status button 구현

Elllin committed:
[FE/#70] Refactor : status button -> Chip으로 변경

Elllin committed:
[FE/#70] Refactor : searchBox, titleBox 높이 맞춤

Elllin committed:
[FE/#70] Feat : InputField props type 추가

Elllin committed:
[FE/#70] Fix : inputField propsType 오류 수정

boolen -> bool로 수정

Elllin committed:
[FE/#70] Refactor : input 값이 있을때만 fetch로직 하도록 구조 변경

Elllin committed:
[FE/#70] Refactor : primaryButton변경사항 취소

color값 전달하려 했으나 필요없어서 원래 상태로 되돌림

sungik-choi committed:
Merge pull request #71 from codesquad-member-2020/fe/feature/detailedIssueHeader

[FE] 이슈 상세 헤더
